src/dspbase.jl

```diff
@@ -787,19 +787,21 @@ end
     conv(u,v,A)
 
 2-D convolution of the matrix `A` with the 2-D separable kernel generated by
-the vectors `u` and `v`.
+the vector `u` and row-vector `v`.
 Uses 2-D FFT algorithm.
 """
-function conv(u::AbstractVector{T}, v::AbstractVector{T}, A::AbstractMatrix{T}) where T
+function conv(u::AbstractVector{T}, v::Transpose{T,<:AbstractVector}, A::AbstractMatrix{T}) where T
```

Comment:
This could also be `Adjoint` instead of `Transpose` (with no change to the implementation below). Would that make more sense? Would one rather separate a separable kernel `H` as `H=u*v'` or `H=u*transpose(v)`? (Anyone around working with complex-valued 2D-data and separable filters?)
Or should that even be `Union{Adjoint{T,<:AbstractVector}, Transpose{T,<:AbstractVector}}` (aka `LinearAlgebra.AdjOrTransAbsVec{T}`)?

Reply:
So I thought I'd ping the author of that method in case he has any insight on whether `Transpose` or `Adjoint` would be more idiomatic. Some `git blame`s later, I've arrived at https://github.com/JuliaLang/julia/commit/823cff9c0714be30a1afb63f412041a73087c08f where @JeffBezanson added this to `signal.j` almost 13 years ago. So Jeff, what do you say?

More seriously, the current code does `transpose`, so the deprecation for `conv(u, v, A)` could either be `conv(u, transpose(v), A)` or `conv(u, conj(v)', A)`, where I find the former more obvious. If we're going with that, it's either `Transpose` or `AdjOrTransAbsVec`, and I tend to prefer the simpler `Transpose` for now. We can always wider the signature later if the need arises.

Another issue that in light of #403, the argument order `conv(A, u, transpose(v))` might be more future-proof, as there, the first argument plays the role of the input and the second one plays the role of a convolution kernel (as inspired by MATLAB).

Reply:
Letting this sink in a bit more, I think these concerns are non-issues:
* As the old code `transpose`d, that's what the deprecation should do. If we want to similarly allow `Adjoint`, we can always do so later.
* If another argument order turns out to be more natural later, we can add support for it then. But I see no reason why we should not support the order now implemented -- convolution is commutative and associative, after all -- so keeping that order for the deprecation seems least surprising.